core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -3684,12 +3684,15 @@ private SqlDialect nonOrdinalDialect() {
     String query = "select cast(\"product_id\" as char) from \"product\"";
     final String expectedMysql = "SELECT CAST(`product_id` AS CHAR)\n"
         + "FROM `foodmart`.`product`";
+    final String expectedMssql = "SELECT CAST([product_id] AS CHAR)\n"
```

Comment:
Add test about if Char with precision.

Reply:
Ok, thank you for helping me review the code, I will modify it